default/python/turn_events/turn_events.py

```diff
@@ -31,4 +31,35 @@ def execute_turn_events():
         if fo.create_field(field_type, x, y, size) == fo.invalid_object():
             print >> sys.stderr, "Turn events: couldn't create new field"
 
+    gsd = fo.get_galaxy_setup_data()
+    # creating monsters
+    monster_freq = fo.monster_frequency(gsd.monsterFrequency)
+    # monster freq ranges from 30 (= one monster per 30 systems) to 3 (= one monster per 3 systems)
+    # (example: low monsters and 150 Systems results in 150 / 30 * 0.001 = 0.005)
+    if monster_freq > 0 and random() < len(systems) / monster_freq * 0.001:
+        #only spawn Krill at the moment, other monsters can follow in the future
+        if random() < 1:
+            monster_type = "SM_KRILL_1"
+        else:
+            monster_type = "SM_FLOATER"
+
+        # search for systems without planets or fleets
+        candidates = [s for s in systems if len(fo.sys_get_planets(s)) <= 0 and len(fo.sys_get_fleets(s)) <= 0]
+        system = choice(candidates)
+        # if system selection fails, report an error
+        if system == fo.invalid_object():
```

Comment:
Sorry again, should've caught that on the first pass, but apparently my mind wasn't working reliably. Anyway: This won't work as intended. `system` will never be `fo.invalid_object()` in this context, in case `candidates` winds up empty (because no systems met the conditions od being empty of both planets and fleets), the `choice` statement will already throw an exception.

What you need to do is to check if candidates actually contains any systems at all, and only proceed if that's the case. And if you got some systems to pick from, you shouldn't have to check for `invalid_object`, as the list should only contain valid system ids (of course, you can decide to be extra safe).